### PR TITLE
Fixes the Mono.Posix API breakage introduced in #3985

### PR DESCRIPTION
**Pull request by ntherning**
Also reverts the changes done to `support/map.c`. This is a generated file which shouldn't be changed directly. An alternate fix for the problem in `map.c` on Windows was to make sure `L_SET` et al are defined in `mph.h` which was already done for Solaris.

### CODE REVIEW

**Comment by directhex** [comment]
whitelist

**Comment by ntherning** [comment]
@jonpryor Please have a look. AFAICT the changes don't break anything so if you are happy with this I believe we can merge.

**Comment by jonpryor** [comment]
LGTM.

File: support/sys-stat.c [hunk 1]
```diff
@@ -76,8 +80,10 @@ Mono_Posix_ToStat (void *_from, struct Mono_Posix_Stat *to)
 	to->st_gid        = from->st_gid;
 	to->st_rdev       = from->st_rdev;
 	to->st_size       = from->st_size;
-	to->st_blksize    = from->st_blksize;
-	to->st_blocks     = from->st_blocks;
+#ifndef HOST_WIN32
+	to->st_blksize = from->st_blksize;
```
Comment (jonpryor) [inline support/sys-stat.c]: ...though it would be nice if you didn't change the vertical alignment of the `=`s here. ;-)

Comment (ntherning) [inline support/sys-stat.c]: I have pushed an update which fixes the vertical alignment.

**Pull request merged**